docs/data/sdk-plugins.md

```diff
@@ -403,4 +403,95 @@ Use a Destination Plugin to send events to a third-party APIs
+
+    ???code-example "Send to Google Tag Manager by pushing events onto the [data layer](https://developers.google.com/tag-platform/tag-manager/web/datalayer) (click to expand)"
+
+        === "JavaScript"
+  
+            ```js
+              import { PluginType } from "@amplitude/analytics-types"
+              export class GTMPlugin {
+                name = "google-tag-manager"
+                type = PluginType.DESTINATION
+
+                constructor(containerId) {
+                  this.containerId = containerId
+                }
+
+                injectGtmScript(containerId) {
+                  if (!window.dataLayer) {
+                    window.dataLayer = window.dataLayer || []
+                    window.dataLayer.push({
+                      "gtm.start": new Date().getTime(),
+                      event: "gtm.js"
+                    })
+                    const head = document.getElementsByTagName("head")[0],
+                      script = document.createElement("script"),
+                      dataLayer = "datalayer" != "dataLayer" ? "&l=" + "datalayer" : ""
+                    script.async = true
+                    script.src =
+                      "https://www.googletagmanager.com/gtm.js?id=" + containerId + dataLayer
+                    head.insertBefore(script, head.firstChild)
+                  }
+                }
+
+                async setup() {
+                  this.injectGtmScript(this.containerId)
+                }
+
+                async execute(event) {
+                  this.injectGtmScript(this.containerId)
+                  window.dataLayer.push(event)
+
+                  return {
+                    code: 0,
+                    event: event,
+                    message: "Event pushed onto GTM Data Layer"
+                  }
+                }
+              }
+            ```
+        
+        === "TypeScript"
+        
+            ```ts
+            import { DestinationPlugin, Event, PluginType, Result } from '@amplitude/analytics-types';
+            export class GTMPlugin implements DestinationPlugin {
+              name = 'google-tag-manager';
+              type = PluginType.DESTINATION as const;
+              containerId: string;
+
+              constructor(containerId: string) {
+                this.containerId = containerId;
+              }
+
+              injectGtmScript(containerId: string) {
+                if (!window.dataLayer) {
+                  window.dataLayer = window.dataLayer || [];
+                  window.dataLayer.push({ 'gtm.start': new Date().getTime(), event: 'gtm.js' });
+                  const head = document.getElementsByTagName('head')[0],
+                    script = document.createElement('script'),
+                    dataLayer = 'datalayer' != 'dataLayer' ? '&l=' + 'datalayer' : '';
+                  script.async = true;
+                  script.src = 'https://www.googletagmanager.com/gtm.js?id=' + containerId + dataLayer;
```

Comment:
reference the installation script below, it seems that we don't need `dataLayer` variable here
```
// <!-- Google Tag Manager -->
<script>
(function (w, d, s, l, i) {
  w[l] = w[l] || [];
  w[l].push({ "gtm.start": new Date().getTime(), event: "gtm.js" });
  var f = d.getElementsByTagName(s)[0],
    j = d.createElement(s),
    dl = l != "dataLayer" ? "&l=" + l : "";
  j.async = true;
  j.src = "https://www.googletagmanager.com/gtm.js?id=" + i + dl;
  f.parentNode.insertBefore(j, f);
})(window, document, "script", "dataLayer", "GTM-TH7PQD7");
</script>
// <!-- End Google Tag Manager -->
```
```suggestion
                  const head = document.getElementsByTagName('head')[0],
                    script = document.createElement('script');
                  script.async = true;
                  script.src = 'https://www.googletagmanager.com/gtm.js?id=' + containerId;
```

Reply:
For purposes of sending events from the SDK -> GTM I believe a datalayer is necessary. It's the way messages from apps are sent to GTM, otherwise those tags can't interact dynamically with the app

[Related docs on datalayer](https://developers.google.com/tag-platform/devguides/datalayer)

Reply:
why do we just 
```
dataLayer = '&l=datalayer';
```
as `'datalayer' != 'dataLayer' ` is true anyway

Reply:
Ah yes that's my bad. I translated the IIFE that their [docs](https://developers.google.com/tag-platform/tag-manager/web#standard_web_page_installation) gave to more human-readable code. I ended up